packages/components/nodes/documentloaders/Zendesk/Zendesk.ts

```diff
@@ -0,0 +1,470 @@
+
+        // Process each locale
+        for (const locale of config.locales) {
+            const articles = await this.fetchAllArticles(locale, config.brandId || undefined, config, axiosHeaders)
+            // Transform each article to the required format
+            for (const article of articles) {
+                const transformedChunks = this.transformArticle(article, config, locale)
+                // Process each chunk (will be 1 chunk for small articles)
+                for (const chunk of transformedChunks) {
+                    allTransformedArticles.push(chunk)
+                }
+            }
+        }
+
+        return allTransformedArticles
+    }
+
+    async init(nodeData: INodeData, _: string, options: ICommonObject): Promise<any> {
+        const zendeskDomain = nodeData.inputs?.zendeskDomain as string
+        const brandId = nodeData.inputs?.brandId as string
+        const localeInputRaw = nodeData.inputs?.locale as string
+        const localeInput = (localeInputRaw && localeInputRaw.trim()) || 'en-us'
+        const publishedArticlesOnly = (nodeData.inputs?.publishedArticlesOnly as boolean) ?? true
+        const charsPerToken = (nodeData.inputs?.charsPerToken as number) ?? 4
+        const metadata = nodeData.inputs?.metadata
+        const _omitMetadataKeys = nodeData.inputs?.omitMetadataKeys as string
+        const output = nodeData.outputs?.output as string
+
+        let omitMetadataKeys: string[] = []
+        if (_omitMetadataKeys) {
+            omitMetadataKeys = _omitMetadataKeys.split(',').map((key) => key.trim())
+        }
+
+        // Parse comma-separated locales
+        const locales = localeInput
+            .split(',')
+            .map((loc) => loc.trim())
+            .filter((loc) => loc.length > 0)
+        
+        // Ensure at least one locale
+        if (locales.length === 0) {
+            locales.push('en-us')
+        }
+
+        const credentialData = await getCredentialData(nodeData.credential ?? '', options)
+        const user = getCredentialParam('user', credentialData, nodeData)
+        const token = getCredentialParam('token', credentialData, nodeData)
+
+        // Build configuration
+        const config: ZendeskConfig = {
+            zendeskDomain,
+            user,
+            token,
+            brandId,
+            publishedArticlesOnly,
+            locales,
+            charsPerToken,
+            api: {
+                protocol: 'https://',
+                helpCenterPath: '/api/v2/help_center/',
+                articlesEndpoint: 'articles.json',
+                publicPath: '/hc/'
+            },
+            defaultTitle: 'Untitled',
+            chunking: {
+                maxTokens: 3000,
+                chunkSize: 1000,
+                overlap: 200
+            }
+        }
+
+        // Validate configuration
+        this.validateConfig(config)
+
+        // Extract articles
+        let docs: IDocument[] = await this.extractAllArticles(config)
```

Comment:
we don't extract all articles in the `init` function, you should create a class that extends `BaseDocumentLoader`, refer to [Airtable.ts](https://github.com/FlowiseAI/Flowise/blob/main/packages/components/nodes/documentloaders/Airtable/Airtable.ts)